### PR TITLE
refactor: database structure, queries and naming

### PR DESCRIPTION
**Pull request by nitrou5**
## Description
- refactor: database structure, queries and naming
- plant gender values can be male or female (changed from man and woman)
- added player near plant checks to server side
 - updated events name prefix to qbx_weed
 
## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -47,9 +47,7 @@ end
 local function updatePlantStats()
     if not (insideHouse or plantsSpawned) then return end
     for k, v in pairs(housePlants[currentHouse]) do
-        local gender = 'M'
-        if v.gender == 'woman' then gender = 'F' end
-
+        local gender = (v.gender == 'female') and 'F' or 'M'
```
Comment (FjamZoo) [inline client/main.lua]:
```suggestion
        local gender = v.gender == 'female' and 'F' or 'M'
```

We don't need the brackets here

File: server/main.lua [hunk 2]
```diff
@@ -1,62 +1,60 @@
...
-    MySQL.query.await('DELETE FROM house_plants WHERE plantid = ? AND building = ?', { plantId, building })
-    TriggerClientEvent('qb-weed:client:refreshHousePlants', -1, building)
+RegisterNetEvent('qbx_weed:server:removeDeathPlant', function(building, plantId, plantCoords)
+    local src = source
+    local player = exports.qbx_core:GetPlayer(src)
+    if not player or not building then return end
+    if #(GetEntityCoords(GetPlayerPed(src)) - plantCoords) > 2 then return end
+    MySQL.prepare.await('DELETE FROM house_plants WHERE id = ?', { plantId })
+    TriggerClientEvent('qbx_weed:client:refreshHousePlants', -1, building)
 end)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
    local player = exports.qbx_core:GetPlayer(source)
    if not player or not building then return end
    if #(GetEntityCoords(GetPlayerPed(player.PlayerData.source)) - plantCoords) > 2 then return end
    MySQL.prepare.await('DELETE FROM house_plants WHERE id = ?', { plantId })
    TriggerClientEvent('qbx_weed:client:refreshHousePlants', -1, building)
end)
```
Here we want to get rid of these local source variables, they are unneeded as soon as we grab the Player object as its already cached there.

File: server/main.lua [hunk 3]
```diff
@@ -73,109 +71,85 @@ local function growPlant(plant)
...
     TriggerClientEvent('qb-weed:client:foodPlant', source, item)
 end)
 
-RegisterServerEvent('qb-weed:server:removeSeed', function(itemslot, seed)
+RegisterServerEvent('qbx_weed:server:removeSeed', function(itemslot, seed)
     local src = source
     local player = exports.qbx_core:GetPlayer(src)
     if not player then return end
 
     player.Functions.RemoveItem(seed, 1, itemslot)
```
Comment (FjamZoo) [inline server/main.lua]:
```suggestion
    local player = exports.qbx_core:GetPlayer(source)
    if not player then return end

    player.Functions.RemoveItem(seed, 1, itemslot)
```

File: server/main.lua [hunk 4]
```diff
@@ -73,109 +71,85 @@ local function growPlant(plant)
...
 end)
 
-RegisterNetEvent('qb-weed:server:harvestPlant', function(house, amount, plantName, plantId)
+RegisterNetEvent('qbx_weed:server:harvestPlant', function(house, seedAmount, plantItemName, plantId, plantCoords)
     local src = source
```
Comment (FjamZoo) [inline server/main.lua]: Get rid of the source variable here aswell

File: server/main.lua [hunk 5]
```diff
@@ -73,109 +71,85 @@ local function growPlant(plant)
...
-    player.Functions.AddItem('weed_' .. plantName, sndAmount)
-    player.Functions.RemoveItem('empty_weed_bag', sndAmount)
-    MySQL.query.await('DELETE FROM house_plants WHERE plantid = ? AND building = ?', { plantId, house })
-    exports.qbx_core:Notify(src, locale('text.the_plant_has_been_harvested'), 'success')
-    TriggerClientEvent('qb-weed:client:refreshHousePlants', -1, house)
 end)
 
-RegisterNetEvent('qb-weed:server:foodPlant', function(house, amount, plantName, plantId)
+RegisterNetEvent('qbx_weed:server:foodPlant', function(house, amount, plantName, plantId)
     local src = source
```
Comment (FjamZoo) [inline server/main.lua]: Get rid of the source variable here aswell

**Review by FjamZoo — CHANGES_REQUESTED** [review]

**Review by FjamZoo — APPROVED** [review]

**Review by mafewtm — APPROVED** [review]

**Pull request merged**